File: ce/customer-engagement/on-premises/snippets/visualbasic/CRMV8/solutionsvb/vb/workwithsolutions10.vb
' <snippetworkwithsolutions10>


 ' Delete a solution

 Dim queryImportedSolution As QueryExpression =
  New QueryExpression With {
   .EntityName = solution.EntityLogicalName,
   .ColumnSet = New ColumnSet(New String() {"solutionid", "friendlyname"}),
   .Criteria = New FilterExpression()
  }


 queryImportedSolution.Criteria.AddCondition("uniquename", ConditionOperator.Equal, ImportedSolutionName)

 Dim ImportedSolution As Solution =
  CType(_serviceProxy.RetrieveMultiple(queryImportedSolution).Entities(0), Solution)

 _serviceProxy.Delete(solution.EntityLogicalName, CType(ImportedSolution.SolutionId, Guid))

 Console.WriteLine("Deleted the {0} solution.", ImportedSolution.FriendlyName)

' </snippetworkwithsolutions10>

File: ce/customerengagement/on-premises/snippets/csharp/CRMV8/solutions/cs/workwithsolutions10.cs
// <snippetworkwithsolutions10>


 // Delete a solution

 QueryExpression queryImportedSolution = new QueryExpression
 {
     EntityName = Solution.EntityLogicalName,
     ColumnSet = new ColumnSet(new string[] { "solutionid", "friendlyname" }),
     Criteria = new FilterExpression()
 };


 queryImportedSolution.Criteria.AddCondition("uniquename", ConditionOperator.Equal, ImportedSolutionName);

 Solution ImportedSolution = (Solution)_serviceProxy.RetrieveMultiple(queryImportedSolution).Entities[0];

 _serviceProxy.Delete(Solution.EntityLogicalName, (Guid)ImportedSolution.SolutionId);

 Console.WriteLine("Deleted the {0} solution.", ImportedSolution.FriendlyName);

// </snippetworkwithsolutions10>